app/helpers/automated_tests_helper.rb

```diff
@@ -320,7 +322,19 @@ def add_feedback_data(results, settings_id, test_id, autotest_setting)
     end
   end
 
-  private
+  # Returns an array of arrays containing a group id and group name to the
+  # starter file entry paths assigned to the grouping associated with the +assignment+
```

Comment:
Okay so everything is good except here I'm pretty sure we need the starter file group for each entry, as multiple groups can have the same entry path. (E.g. when the "assign one starter file group randomly" option is selected)

The starter file mapping csv download method uses a helper that generates both group name and entry path, so maybe you could reuse that (or adapt the code here, I don't really care)